compiler/rustc_lint/messages.ftl

```diff
@@ -406,6 +406,14 @@ lint_non_fmt_panic_unused =
     }
     .add_fmt_suggestion = or add a "{"{"}{"}"}" format string to use the message literally
 
+lint_non_local_definitions = non local definition should be avoided as they go against expectation
+    .help =
+        move this declaration outside the of the {$depth ->
+            [one] current body
+           *[other] {$depth} outermost bodies
```

Comment:
```suggestion
            [one] current function body
           *[other] {$depth} outermost function bodies
```
I feel like users may be confused by "body", I think this is compiler dev jargon.

Reply:
I've made the error message use the `DefKind`, so we can show "constant", "function", ...